Sharing saved-objects phase 1.5 (#75444)
elastic/kibana: docs/api/saved-objects/resolve_import_errors.asciidoc

    nothing.
    `destinationId`:::
    (Optional, string) Specifies which destination ID the imported object should have (if different from the current ID).
    `replaceReferences`:::
    (Optional, array) A list of `type`, `from`, and `to` used to change the object references.
    `ignoreMissingReferences`:::
    (Optional, boolean) When set to `true`, any missing references errors are ignored. When set to `false`, this does nothing.
=====

[[saved-objects-api-resolve-import-errors-response-body]]
==== Response body

`success`::
  (boolean) Indicates if the import was completely successful. When set to `false`, some objects may have been copied. For additional
  information, refer to the `errors` and `successResults` properties.

`successCount`::
  (number) Indicates the number of successfully resolved records.

`errors`::
  (Optional, array) Specifies the objects that failed to resolve.
+
NOTE: One object may result in multiple errors which require separate steps to resolve (for instance, a `missing_references` error and a
`conflict` error).

`successResults`::

>   (Optional, array) Indicates the objects that were imported successfully, with any metadata if applicable.

Reviewer: ```suggestion
  (Optional, array) Indicates the objects that are successfully imported, with any metadata if applicable.
```

+

> NOTE: No objects are actually created until all resolvable errors have been addressed! This includes conflict errors and missing references

Reviewer: ```suggestion
NOTE: Objects are only created when errors are resolved, which includes conflict and missing references
```

> errors. See the examples below for how to resolve these errors.

Reviewer: ```suggestion
errors. To resolve errors, refer to the <<saved-objects-api-resolve-import-errors-example, examples>>.
```

[[saved-objects-api-resolve-import-errors-example]]
==== Examples

[[saved-objects-api-resolve-import-errors-example-1]]

> ===== 1. Resolve conflict errors

Reviewer: ```suggestion
===== Resolve conflict errors
```

This example builds upon the <<saved-objects-api-import-example-3,Import objects API example with conflict errors>>.

> Resolve conflict errors for an index pattern, visualization, and canvas by overwriting the existing saved objects:

Reviewer: ```suggestion
Resolve conflict errors for an index pattern, visualization, and *Canvas* workpads by overwriting the existing saved objects:
```

--------------------------------------------------
{
  "success": true,
  "successCount": 4,
  "successResults": [
    {
      "id": "my-pattern",
      "type": "index-pattern",
      "meta": {
        "icon": "indexPatternApp",
        "title": "my-pattern-*"
      }
    },
    {
      "id": "my-vis",
      "type": "visualization",
      "destinationId": "another-vis",
      "meta": {
        "icon": "visualizeApp",
        "title": "Look at my visualization"
      }
    },
    {
      "id": "my-canvas",
      "type": "canvas-workpad",
      "destinationId": "yet-another-canvas",
      "meta": {
        "icon": "canvasApp",
        "title": "Look at my canvas"
      }
    },
    {
      "id": "my-dashboard",
      "type": "dashboard",
      "meta": {
        "icon": "dashboardApp",
        "title": "Look at my dashboard"
      }
    }
  ]
}
--------------------------------------------------

> This result indicates that the import was successful, and all four objects were created.

Reviewer: ```suggestion
The result indicates a successful import, and all four objects were created.
```

TIP: If a prior import attempt resulted in resolvable errors, you must include a retry for each object you want to import, including any
that were returned in the `successResults` array. In this example, we retried importing the dashboard accordingly.

[[saved-objects-api-resolve-import-errors-example-2]]

> ===== 2. Resolve missing reference errors

Reviewer: ```suggestion
===== Resolve missing reference errors
```

--------------------------------------------------
{
  "success": true,
  "successCount": 3,
  "successResults": [
    {
      "id": "my-vis",
      "type": "visualization",
      "meta": {
        "icon": "visualizeApp",
        "title": "Look at my visualization"
      }
    },
    {
      "id": "my-search",
      "type": "search",
      "meta": {
        "icon": "searchApp",
        "title": "Look at my search"
      }
    },
    {
      "id": "my-dashboard",
      "type": "dashboard",
      "meta": {
        "icon": "dashboardApp",
        "title": "Look at my dashboard"
      }
    }
  ]
}
--------------------------------------------------

> This result indicates that the import was successful, and all three objects were created.

Reviewer: ```suggestion
The result indicates a successful import, and all three objects were created.
```
